src/main/java/com/google/api/codegen/ruby/RubyGapicContext.java

```diff
@@ -342,20 +345,35 @@ public String rubyTypeName(TypeRef typeRef) {
     return Splitter.on("::").splitToList(getApiConfig().getPackageName());
   }
 
-  public DynamicLangApiMethodView getMethodView(Interface service, Method method) {
+  public OptionalArrayMethodView getMethodView(Interface service, Method method) {
+    SurfaceTransformerContext context = getSurfaceTransformerContextFromService(service);
+    MethodTransformerContext methodContext = context.asMethodContext(method);
+    ApiMethodTransformer methodTransformer = new ApiMethodTransformer();
+    return methodTransformer.generateOptionalArrayMethod(methodContext);
```

Comment:
At this point I think we should rename generateOptionalArrayMethod to generateDynamicLangApiMethod.